src/OpenTelemetry/Trace/BatchingActivityProcessor.cs

```diff
@@ -320,14 +320,18 @@ private async Task<int> ExportBatchAsync(CancellationToken cancellationToken)
                     this.batch.Add(nextActivity);
                 }
 
-                var result = await this.exporter.ExportAsync(this.batch, cancellationToken).ConfigureAwait(false);
-                if (result != ExportResult.Success)
+                using (Sdk.SuppressInstrumentation)
```

Comment:
@CodeBlanch @reyang 

Uh oh this is a terrible bug waiting to happen. This should be:
```
using (Sdk.SuppressInstrumentation)
```

This bug would result in the static instance getting disposed. I think we need to rethink things a bit to prevent this from being possible.

Reply:
+100